subprojects/core/src/test/groovy/org/gradle/internal/vfs/LinuxFileWatcherUpdaterTest.groovy

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs
+
+import net.rubygrapefruit.platform.file.FileWatcher
+import org.gradle.internal.vfs.watch.FileWatcherUpdater
+
+class LinuxFileWatcherUpdaterTest extends AbstractFileWatcherUpdaterTest {
+
+    @Override
+    FileWatcherUpdater createUpdater(FileWatcher watcher) {
+        new LinuxFileWatcherUpdater(watcher)
+    }
+
+    def "adds watches for all must watch directories"() {
+        def mustWatchDirectoryRoots = ["first", "second"].collect { file(it).createDir()}
+        def mustWatchDirectories = mustWatchDirectoryRoots + file("first/within").createDir()
```

Comment:
Nit: splitting this into two separate variables does not seem to add anything for me, can we merge them?

Reply:
I am using the separate variables, so I can easily assert which parts we should watch and which we shouldn't.

Reply:
Alright.